.github/workflows/test.yml

```diff
@@ -102,58 +103,98 @@ jobs:
         elif echo "$GITHUB_REF" | grep -q '^refs/tags/'; then
           # Pushes tag - deploy tag name.
           echo "TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+        else
+          # Use commit SHA for PR and forks
+          echo "TAG=${GITHUB_SHA::8}" >> $GITHUB_ENV
         fi
-        echo "DOCKER_BASE=${{ secrets.DOCKER_USER }}/${{ env.DOCKER_REPO }}" >> $GITHUB_ENV
+        [[ -n "${{ env.DOCKER_USER }}" ]] || DOCKER_USER="test"
```

Comment:
Nit: What if you just defined `DOCKER_TEST_USER="test"` and used that instead of `DOCKER_USER` during build and test steps? Only the 'skopeo copy' really needs the actual `DOCKER_USER` on the destination repository.